test/test_StateMachineDevice.py

```diff
@@ -35,6 +35,17 @@ class MockStateMachineDevice(StateMachineDevice):
 
 
 class TestStateMachineDevice(unittest.TestCase):
+    def test_not_implemented_errors(self):
+        self.assertRaises(NotImplementedError, StateMachineDevice)
+
+        with patch('lewis.devices.StateMachineDevice._get_state_handlers',
+                   return_value={'test': Mock()}):
+            self.assertRaises(NotImplementedError, StateMachineDevice)
+
+            with patch('lewis.devices.StateMachineDevice._get_initial_state',
+                       return_value='test'):
+                self.assertRaises(NotImplementedError, StateMachineDevice)
```

Comment:
I guess it's a question of what exactly the test is trying to guarantee about framework behaviour. My guess (which may be wrong) is that you're trying to prove/ensure that...

> This base class has methods A, B and C that must be implemented by derived classes. 
>
> Construction fails if __any__ of A, B and C are not implemented. 
>
> Construction succeeds if __all__ of A, B and C are implemented.

In the first case my complaint is that you're not really testing that construction fails if __any__ of A, B or C are not implemented. Because the test will still pass if A and B do not throw such exceptions. Actually, it will also pass if C doesn't throw such an exception, but some other X does.

In the second case I'm saying there is no test for successful construction when __all__ of A, B and C are implemented.

To test the conditions I quoted, you'd need to do something like:

```python
def test(...):
    class CaseAMissing(BaseClass):
        B = None
        C = None
    self.assertRaises(NotImplementedError, CaseAMissing)

    class CaseBMissing(BaseClass):
        A = None
        C = None
    # ...
    # ...
    class CaseAllImplemented(BaseClass):
        # A, B, C = None
    assertRaisesNothing(self, CaseAllImplemented)
```
Although that's a very brute force approach... maybe there's a cleaner way of implementing that.